pkg/cli/pid.go

```diff
@@ -3,52 +3,80 @@
 package cli
 
 import (
+	"errors"
 	"fmt"
 	"os"
-	"syscall"
+	"time"
 
 	"github.com/replicate/pget/pkg/logging"
 )
 
 type PIDFile struct {
+	Path string
 	file *os.File
 	fd   int
 }
 
-func NewPIDFile(path string) (*PIDFile, error) {
-	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0644)
+func (p *PIDFile) tryCreateLockFile(path string) (*os.File, error) {
+	logger := logging.GetLogger()
+
+	lockedFile, err := os.OpenFile(p.Path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0644)
+
 	if err != nil {
-		return nil, err
+		if errors.Is(err, os.ErrExist) {
```

Comment:
You log the error here but then you return the lock file with no error  so the caller has no idea that this failed.